### DISCUSSION TITLE
Shares - Dynamic choose for related items

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#11054] [Feature Requests]
Hello everyone, has anyone thought about the possibility of being able to select which related item we want to be shared, when we share an item? I have a series of practical cases where this would be perfect :)

**Comment by Jeyakumarrathnaselvi** [comment]
Any reply appreciated pls

**Reply by rijkvanzanten** [reply]
What sort of reply? I think it's a good idea, and that's why I upvoted it 🙂

**Comment by xu4wang** [comment]
For current implementation, I open an item and then I can click share, so the share is a feature when a specific item is already selected...

Do you mean have another feature to initiate share first, before an item is open(selected)? @hugopvl

**Reply by hugopvl** [reply]
@xu4wang 
What I mean is something like this: in a first share I just want to share the item + TRF-7132, in the second share I want to share the item + TRF-7133,for example, I've thought of several solutions, through roles and permissions, but in this type of situations the ID of the share, it has to be included :)

![image](https://user-images.githubusercontent.com/81326296/158426655-b10ae291-8e3d-4be4-b5aa-364f77753138.png)

**Reply by xu4wang** [reply]
I think it's about to config the access permission of the two items.

create one role who can read 7132 and another role who can read 7133, and create two shares for the above two roles. Will that work?

@hugopvl

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
